lms/services/canvas_api/client.py

```diff
@@ -226,16 +227,29 @@ class _ListFilesSchema(RequestsResponseSchema):
         id = fields.Integer(required=True)
         updated_at = fields.String(required=True)
 
-    def public_url(self, file_id):
+    def public_url(self, course_id, file_id, is_instructor):
```

Comment:
I added `course_id` and `is_instructor` as arguments to this method.

`is_instructor` could have been added as an argument to `Client.__init__()` and then the service factory would pass in `request.lti_user.is_instructor`. But that would make `is_instructor` potentially a dependency of all of `Client`'s methods. Putting it here clarifies that only this one method uses it

Reply:
We could do this check in the LTI launch instead, e.g. in [`canvas_file_basic_lti_launch()`](https://github.com/hypothesis/lms/blob/067943f29fd8b880566bd6be51fbee0e485e86db/lms/views/basic_lti_launch.py#L74-L90). Then we wouldn't need to add these params to `public_url`. That would break a pattern in the LMS app's code architecture though: all third-party API requests are made during proxy API requests, not during LTI launch requests. The main reason, though, is that by adding this to the proxy API call we can trigger the proxy API to return an error response with an error code, and the frontend already has a way to trigger an error dialog in response to an error code from the proxy API. If we did the check during the LTI launch instead, it'd be more work to make the frontend display an error message